packages/ui-components/src/Input/Field.tsx

```diff
@@ -1,70 +1,42 @@
-import React, { FunctionComponent, useState } from "react";
+import React from "react";
 import { Field, FieldProps, FieldRenderProps } from "react-final-form";
-
 import {
   SingleLineTextInput,
-  TextInputProps,
+  MultilineTextInput,
   NumberInput,
-  NumberProps,
 } from "./TextTypeInput";
-import {
-  EmailInput,
-  NewPasswordInput,
-  ExistingPasswordInput,
-  EmailProps,
-  NewPasswordProps,
-  ExistingPasswordProps,
-  PasswordInputType,
-} from "./EmailPasswordInput";
-import { CheckboxInput, CheckboxInputProps } from "./CheckboxInput";
-import Icon from "../Icon/Icon";
-
-// the following omit statements can be removed
-// once type better limits what props can be passed in
-type TextInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    TextInputProps,
-  "type"
->;
-
-type NumberInputFieldProps = Omit<
-  FieldProps<number, FieldRenderProps<number, HTMLElement>, HTMLElement> &
-    NumberProps,
-  "type"
->;
-
-type CheckboxInputFieldProps = Omit<
-  FieldProps<boolean, FieldRenderProps<boolean, HTMLElement>, HTMLElement> &
-    CheckboxInputProps,
-  "type"
->;
-
-type EmailInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    EmailProps,
-  "type"
->;
-
-type NewPasswordInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    NewPasswordProps,
-  "type"
->;
-
-type ExistingPasswordInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    ExistingPasswordProps,
-  "type"
->;
-
-type InternalFieldProps =
-  | ({ InputFieldComponent: React.FunctionComponent } & TextInputFieldProps)
-  | CheckboxInputFieldProps
-  | EmailInputFieldProps
-  | NewPasswordInputFieldProps
-  | ExistingPasswordInputFieldProps;
-
-const InputField = ({
+import { EmailInput, PasswordInput } from "./EmailPasswordInput";
+import { CheckboxInput } from "./CheckboxInput";
+
+type InputComponentTypes =
+  | typeof PasswordInput
+  | typeof EmailInput
+  | typeof SingleLineTextInput
+  | typeof MultilineTextInput
+  | typeof NumberInput
+  | typeof CheckboxInput;
+
+// Extends field render props, which will be passed to the Field component in InputField
+export interface InputFieldProps<
+  IComponent extends React.ComponentType<InputComponentTypes>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+> extends FieldProps<any, FieldRenderProps<any, HTMLElement>, HTMLElement> {
+  as: IComponent;
+  // should include types currently supported
+  type: "text" | "email" | "password" | "number" | "checkbox";
+}
+
+// remove any properties defined in InputFieldProps, so we can handle them exclusively.
+type FieldPropsRefined<
+  IComponent extends React.ComponentType<InputComponentTypes>
+> = InputFieldProps<IComponent> &
+  Omit<
+    React.ComponentPropsWithoutRef<IComponent>,
+    keyof InputFieldProps<IComponent>
+  >;
+
+export function InputField({
```

Comment:
Shouldn't this file be named `InputField` as we generally name the file after the default or primary export? Also this should probably be the `default` export.

Reply:
Okay I can rename the file to `InputField`, since we can't name this component `Field` since it renders a child component `Field` as well.